syntax = "proto3";

package catalogue_proto;

message Bus {
	string name = 1;
	double factual_route_length = 2;
	double geo_route_length = 3;
	double curvature = 4;
	int32 unique_stops_count = 5;
	repeated int32 stop_index = 6;
	bool is_cycled = 7;	
}

message Stop {
	string name = 1;
	double lattitude = 2;
	double longtitude = 3;
	repeated int32 bus_index = 4;
}

message Point {
	double x = 1;
	double y = 2;
}

message RGB {
	int32 red = 1;
	int32 green = 2;
	int32 blue = 3;
}

message RGBA {
	int32 red = 1;
	int32 green = 2;
	int32 blue = 3;
	double opacity = 4;	
}

message Color {
	RGB rgb = 1;
	RGBA rgba = 2;
	string str = 3;
}

message RenderSettings {
	double width = 1;
	double height = 2;
	double padding = 3;
	double line_width = 4;
	double stop_radius = 5;
	int32 bus_label_font_size = 6;
	Point bus_label_offset = 5;
	int32 stop_label_font_size = 6;
	Point stop_label_offset = 7;
	Color underlayer_color = 8;
	double underlayer_width = 9;
	repeated Color color_palette = 10;
}

message TransportCatalogue {
	RenderSettings render_settings = 1;
	repeated Bus bus = 3;
	repeated Stop stop = 4;
}